### PR TITLE
Add replay option to disable deadlock detection

### PR DESCRIPTION
**Pull request by jlegrone**
## What was changed

This PR adds a new field `DisableDeadlockDetection` to `WorkflowReplayerOptions`. Setting this to `true` disables deadlock detection during replay.

## Why?

The `TEMPORAL_DEBUG` environment variable is intended to serve a similar purpose, but can be [tricky to configure](https://github.com/temporalio/sdk-go/issues/329#issuecomment-1464894075) and is a global setting.

Deadlock detection can be disabled programmatically using `DeadlockDetectionTimeout` on `worker.Options`. It makes sense to support something similar in `WorkflowReplayerOptions` since attaching a debugger is such a common use case when replaying workflows.

### How was this tested:

Please see included unit test.

### Any docs updates needed?

The updated godoc is probably enough, though this would be a useful feature to mention in tutorials about workflow debugging.

### CODE REVIEW

File: internal/internal_worker.go [hunk 1]
```diff
@@ -1370,6 +1376,9 @@ func (aw *WorkflowReplayer) replayWorkflowHistory(logger log.Logger, service wor
 			SdkMetadata:                     true,
 		},
 	}
+	if aw.disableDeadlockDetection {
+		params.DeadlockDetectionTimeout = math.MaxInt64
```
Comment (cretz) [inline internal/internal_worker.go:1380]:
```suggestion
		params.DeadlockDetectionTimeout = unlimitedDeadlockDetectionTimeout
```

Same thing, but we already have a constant for this. But not a merge blocker, just if you make another commit.

**Review by cretz — APPROVED** [review]
Yeah, we had traditionally avoided being able to disable deadlock detection programmatically in the Go SDK specifically because it can/will be misused. And we're supporting "debug mode" in other SDKs via parameter and env var too, so I suppose we can here.

Will wait to merge to get @Quinn-With-Two-Ns's approval

**Review by Quinn-With-Two-Ns — APPROVED** [review]

**Comment by cretz** [comment]
@jlegrone - merging master into your branch and merging when CI passes
